### PR TITLE
Replace all date formatting to use native APIs

### PR DESCRIPTION
**Pull request by deanveloper**
also my IDE did some linting... not sure if this goes against style guide, i'll see if i can fix this in a second commit if needed

### CODE REVIEW

**Comment by deanveloper** [comment]
is there an easy way to test these changes btw? no clue how to run stuff locally without going through the entire mastodon setup 😅

**Comment by deanveloper** [comment]
actually, maybe making this configurable in `glitch-soc` settings is the best way to do it. i'll make some more changes, as well as undoing my automatic linter changes with all the semicolons and spaces to hopefully clean up the diff a bit

**Comment by neatchee** [comment]
Yeah, I'd definitely like to leave style changes to upstream. Less likely to have our oil requests rejected

File: app/javascript/flavours/glitch/features/account/components/header.js [hunk 1]
```diff
@@ -167,11 +167,11 @@ class Header extends ImmutablePureComponent {
...
-    let info        = [];
-    let actionBtn   = '';
-    let bellBtn     = '';
-    let lockedIcon  = '';
-    let menu        = [];
+    let info       = [];
+    let actionBtn  = '';
+    let bellBtn    = '';
+    let lockedIcon = '';
+    let menu       = [];
```
Comment (deanveloper) [inline app/javascript/flavours/glitch/features/account/components/header.js:174]: still some minor spacing issues... i'll fix it LOL

**Comment by deanveloper** [comment]
Still have more work to do since I want to see if I can make this configurable

**Comment by deanveloper** [comment]
Actually, I think no one really cares about the fact that mastodon is 24-hour time... Might be wrong about that but making it configurable seems like too much effort (for now) for too little reward.

**Comment by deanveloper** [comment]
If this gets merged, can it be squashed so that the commit can be easily cherry-picked for an MR into glitch-soc?

**Comment by neatchee** [comment]
Yup. I'll make sure it gets squashed :)

**Comment by neatchee** [comment]
It appears this whole thing is a bust :( This project still uses react-intl v2.9.0 which doesn't support passing down hourCycle. I'd migrate to a higher version of react-intl but v2 => v3 has significant breaking changes and may require a refactor of anywhere react-intl is used across the project.  Punting this for now until one of us has some time to determine the consequences of that large a change :\

**Comment by deanveloper** [comment]
oh perfect, just found this so i should be able to test my changes :) https://docs.joinmastodon.org/dev/setup/

**Pull request merged**